v5/v5_with_library/v5_stable.ino

```diff
@@ -617,49 +618,53 @@ String getResponse() {
+    } else {
+      newMessageNumber = getNewMessageNumber(response);
+      senderNumber = getMobileNumberOfMsg(String(newMessageNumber), true);
+      _message_ = (removeNewline(removeOk(readMessage(newMessageNumber))));
+    }
+    if (!isIn(senderNumber, "3374888420")) {
       // if message is not send by module it self
-      if (senderIsAuthentic(senderNumber, _message_))
-        _message_ = executeCommand(_message_);
+      String messageState = "";
+      bool authenticSender = senderIsAuthentic(senderNumber, _message_);
+      if (authenticSender)
+        messageState = executeCommand(_message_);
       else
         println("Unauthorize sender, message {" + _message_ + "} not executed");
       println("New message [ " + _message_ + "]");
-      if (_message_.indexOf("<executed>") != -1)
+      if (isIn(messageState, "<executed>"))
         deleteMessage(newMessageNumber, _message_, senderNumber);
       else { // message is not executed
         if (!companyMsg(senderNumber)) {
-          if (newPackageSubscribed(_message_)) {
-            // ~ part not belongs to this condition just for testing
-            // increment 30 days in date and month or just month
-            updatePackageSubscribedDate();
-          }
           sendSMS("<Unable to execute sms no. {" + String(newMessageNumber) +
-                  "} message: > [ " +
-                  removeNewline(_message_.substring(
-                      0, _message_.indexOf(" <not executed>"))) +
+                  "} message: > [ " + removeNewline(_message_) +
                   " ] from: " + senderNumber);
-          toOrangePi("untrained_message:" + removeNewline(_message_) +
-                     " from: {_" + senderNumber + "_}<_" +
-                     String(newMessageNumber) + "_>");
+          authenticSender
+              ? toOrangePi("untrained_message: _[_" + removeNewline(_message_) +
+                           (pending_test_message ? "isDemo" : "") +
+                           "_]_ from: _{_" + senderNumber + "_}_ index: _(_" +
+                           String(newMessageNumber) + "_)_")
+              : log("skipping unauthorized message from: " + senderNumber);
         } else {
           Println(3, "Company message received deleting it...");
-          if (isValidString(_message_))
-            sendSMS(
-                "<Company message received sms no. {" +
-                String(newMessageNumber) + "} message: > [ " +
-                _message_.substring(0, _message_.indexOf(" <not executed>")) +
-                " ] from: " + senderNumber + ". deleting it...");
-          else
+          if (isValidString(_message_)) {
+            sendSMS("<Company message received sms no. {" +
+                    String(newMessageNumber) + "} message: > [ " + _message_ +
+                    " ] from: " + senderNumber + ". deleting it...");
+            if (newPackageSubscribed(_message_))
+              updatePackageSubscribedDate();
```

Comment:
I like this approach, it improve code complexity unreadable company messages will no more executed to test the new packages subscription.